src/creep.prototype.js

```diff
@@ -573,6 +586,122 @@ Creep.prototype.routeTo = function routeCreep(target, opts) {
+          && !curr.structure instanceof StructureRoad
+          && !curr.structure instanceof StructureContainer
+          && !(curr.structure instanceof StructureRampart && (curr.my || (!curr.my && curr.isPublic)))
+        )
+      ) {
+        return false;
+      }
+    }
+    const creepsAtMySpot = newPos.lookFor(LOOK_CREEPS);
+    if (creepsAtMySpot.length) {
+      const creepAtMySpot = creepsAtMySpot[0];
+      if (!creepAtMySpot.my) {
+        return false;
+      } else if (creepAtMySpot.my && !creepsToMove[creepAtMySpot.name]) {
+        // Try to move this creep out of the way....
+        const [target, range] = getTargetAndRange(creepAtMySpot);
+        creepsToMove[creepAtMySpot.name] = { target, range }
+        moveCreepOutOfTheWay(creepAtMySpot, target, range, creepsToMove, nextPositions, creepMoves);
+      }
+    }
+    if (
+      !nextPositions.find(pos => pos.x === newPos.x && pos.y === newPos.y)
+      && !creepMoves.find(([creep, pos]) => pos.x === newPos.x && pos.y === newPos.y)
+    ) {
+      return true;
+    }
+  });
+  if (target) {
+    let newPos;
+    const keepRangeSpot = availableSpots.find(([x, y]) => {
+      newPos = new RoomPosition(x, y, creep.room.name);
+      const newRange = newPos.getRangeTo(target);
+      if (
+         newRange === range
+        || (newRange <= range && newRange > creep.pos.getRangeTo(target))
+      ) {
+        return true;
+      }
+    });
+    if (newPos && keepRangeSpot) {
+      creepMoves.push([creep, newPos]);
+    }
+  } else if (availableSpots.length) {
+    creepMoves.push([creep, availableSpots[0]]);
+  }
+}
+
+Creep.getOutOfTheWay = function getAllCreepsOutOfTheWay() {
+  if (lastTick !== Game.time) {
+    lastTick = Game.time;
+    movingCreeps = {};
+  }
+  const creepsToMove = {};
+  const creepMoves = [];
+  const nextPositions = [];
+  for (let [creepName, { target, nextPosition }] of Object.entries(movingCreeps)) {
+    if (nextPosition) {
+      nextPositions.push(nextPosition);
+      const creep = Game.creeps[creepName];
+      const nextPosCreeps = nextPosition.lookFor(LOOK_CREEPS);
+      if (nextPosCreeps) {
+        const creepsAtNextPos = nextPosCreeps.filter(creep => creep.my);
+        if (creepsAtNextPos.length) {
+          const creepAtNextPos = creepsAtNextPos[0];
+          if (creepAtNextPos !== creep) {
+            if (creepsToMove[creepAtNextPos.name]) {
+              // creep is already trying to move.... ignore
+              continue;
+            }
+            if (!creepsToMove[creepAtNextPos.name]) {
+              const [target, range] = getTargetAndRange(creepAtNextPos);
+              creepsToMove[creepAtNextPos.name] = { target, range };
+            }
+          }
+        }
+      }
+    }
+  }
+  for (let [creepName, { target, range }] of Object.entries(creepsToMove)) {
+    const creep = Game.creeps[creepName];
+    moveCreepOutOfTheWay(creep, target, range, creepsToMove, nextPositions, creepMoves);
```

Comment:
For every creep tagged as needing to move out of the way, call `moveCreepOutOfTheWay`